### PR TITLE
Add Arcadia Configuration

### PR DESCRIPTION
**Pull request by lcadzy**

### CODE REVIEW

**Comment by t-rufang** [comment]
>       </action>

We need Arcadia type here

---
Refers to: PluginsAndFeatures/azure-toolkit-for-intellij/resources/META-INF/plugin.xml:332 in 1984878. [](commit_id = 1984878273d5467aba4908c435a03c086cc2a998, deletion_comment = False)

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkSubmissionContentPanel.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import com.microsoft.azure.hdinsight.spark.ui.ArcadiaSparkClusterListRefreshableCombo
+import com.microsoft.azure.hdinsight.spark.ui.SparkClusterListRefreshableCombo
+import com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionContentPanel
+
+class ArcadiaSparkSubmissionContentPanel (project: Project) : SparkSubmissionContentPanel(project, "Arcadia Spark") {
+    override val clustersSelection: SparkClusterListRefreshableCombo by lazy { ArcadiaSparkClusterListRefreshableCombo().apply {
+        Disposer.register(this@ArcadiaSparkSubmissionContentPanel, this@apply)
+    } }
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkSubmissionContentPanel.kt]: We need to override the `Spark Clusters(Linux only)` label with `Spark Computes`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt [hunk 2]
```diff
@@ -227,3 +230,13 @@ class CosmosServerlessSparkAccountsCombo: SparkClusterListRefreshableCombo() {
 
     override val viewModel: SparkClusterListRefreshableCombo.ViewModel by lazy { ViewModel() }
 }
+
+class ArcadiaSparkClusterListRefreshableCombo: SparkClusterListRefreshableCombo() {
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt]: We need to override `comboBoxName` with `sparkComputeListComboBox` for distinguish different kind of spark cluster telemetries.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitStorageTypeOptionsForCluster.java [hunk 3]
```diff
@@ -75,6 +75,11 @@
             SparkSubmitStorageType.DEFAULT_STORAGE_ACCOUNT
     }),
 
+    // arcadia cluster now suppports blob
+    ArcadiaSparkCluster(new SparkSubmitStorageType[]{
+            SparkSubmitStorageType.BLOB
+    }),
```
Comment (t-rufang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitStorageTypeOptionsForCluster.java]: Maybe we need to specify `getDefaultStorageType` and  `getStorageOptionsType` for class `ArcadiaSparkCompute`.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanel.kt [hunk 4]
```diff
@@ -123,14 +123,16 @@ open class SparkSubmissionContentPanel(private val myProject: Project, val type:
...
+
     open fun getErrorMessageClusterNameNull(isSignedIn: Boolean): String {
         return when {
             isSignedIn -> "Cluster name should not be null, please choose one for submission"
             else -> "Can't list cluster, please login within Azure Explorer (View -> Tool Windows -> Azure Explorer) and refresh"
         }
     }
 
-    val clustersSelectionPrompt: JLabel = JLabel("Spark clusters(Linux only)").apply {
+    val clustersSelectionPrompt: JLabel = JLabel("$clusterHint").apply {
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanel.kt]: `val clustersSelectionPrompt: JLabel = JLabel(clusterHint).apply` might be clear. But never mind, no big deal.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java [hunk 5]
```diff
@@ -49,4 +50,9 @@ public String getConnectionUrl() {
     public SubscriptionDetail getSubscription() {
         return null;
     }
+
+    @Override
+    public SparkSubmitStorageTypeOptionsForCluster getStorageOptionsType() {
+        return SparkSubmitStorageTypeOptionsForCluster.ArcadiaSparkCluster;
+    }
```
Comment (t-rufang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/projectarcadia/common/ArcadiaSparkCompute.java]: Do we also need to override method `getDefaultStorageType`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkSubmissionContentPanel.kt [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    override val clusterHint: String
+        get() = "Spark Computes(Linux only)"
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkSubmissionContentPanel.kt]:
```suggestion
        get() = "Spark Computes"
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt [hunk 7]
```diff
@@ -227,3 +230,15 @@ class CosmosServerlessSparkAccountsCombo: SparkClusterListRefreshableCombo() {
...
+    override val comboBoxName: String = "computeListComboBox"
+
+    inner class ViewModel
+        : SparkClusterListRefreshableCombo.ViewModel(ArcadiaSparkComputeManager.getInstance().clusters
+            .toTypedArray()) {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkClusterListRefreshableCombo.kt]: `.sortedBy { it.title }`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkConfiguration.kt [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    override fun getSparkApplicationType(): SparkApplicationType {
+        return SparkApplicationType.ArcadiaSpark
+    }
+
+    override fun getConfigurationEditor(): SettingsEditor<out RunConfiguration> {
+        return LivySparkRunConfigurationSettingsEditor(ArcadiaSparkConfigurable(module.project))
+    }
+
+    override fun getSuggestedNamePrefix(): String {
+        return "[Spark on Arcadia]"
```
Comment (lcadzy) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/ArcadiaSparkConfiguration.kt]: >  return "[Spark on Arcadia]" [](start = 6, length = 29)

Need to test this ,seems not work

**Pull request merged**